cherryrgb/src/lib.rs

```diff
@@ -122,6 +117,41 @@ pub fn find_devices(product_id: Option<u16>) -> Result<Vec<(u16, u16)>> {
     Ok(usb_ids)
 }
 
+/// Reads the given color profile file and returns a vector of `ProfileKey`.
+/// # Arguments
+/// * `file_path` - Path of the file that holds the color profile information.
+pub fn read_color_profile_file(file_path: &str) -> Result<Vec<ProfileKey>> {
+    let mut f = std::fs::File::open(file_path).context(file_path.to_string())?;
```

Comment:
Rename to `read_color_profile` and make it take a str-ref `color_profile: &str`.

Move the logic of opening the file and reading to string into the CLI main.rs.

This keeps the library function more universal - it shouldn't care where the input comes.